src/interfaces/IHookMetadata.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+interface IHookMetadata {
+    // Struct representing the auditor of a smart contract
+    struct Auditor {
+        string name;       // Name of the auditor
+        string uri;        // URI with additional information about the auditor
+        string[] authors;  // List of authors who are responsible for the audit
+    }
+
+    // Struct representing a summary of the audit
+    struct AuditSummary {
+        Auditor auditor;         // The auditor who performed the audit
+        uint256 issuedAt;        // The timestamp at which the audit was issued
+        uint256[] ercs;          // List of ERC standards that were covered in the audit
+        bytes32 codeHash;        // Hash of the audited smart contract code
+        bytes32 auditHash;       // Hash of the audit document
+        string auditUri;         // URI with additional information or the full audit report
+    }
+
+    // Struct representing the EIP712 domain, which is used for signatures
+    struct EIP712Domain {
```

Comment:
is there a reason not to pre-define the eip712 typehash of what the signature should be of?